Source/interfac.cpp

```diff
@@ -235,6 +239,64 @@ void DrawCutsceneForeground()
 	RenderPresent();
 }
 
+struct {
+	uint32_t loadStartedAt;
+	EventHandler prevHandler;
+	bool skipRendering;
+	bool done;
+	uint32_t drawnProgress;
+	std::array<SDL_Color, 256> palette;
+} ProgressEventHandlerState;
+
+void InitRendering()
+{
+	// Blit the background once and then free it.
+	DrawCutsceneBackground();
+	if (RenderDirectlyToOutputSurface && PalSurface != nullptr) {
+		// Render into all the backbuffers if there are multiple.
+		const void *initialPixels = PalSurface->pixels;
+		if (DiabloUiSurface() == PalSurface)
+			BltFast(nullptr, nullptr);
+		RenderPresent();
+		while (PalSurface->pixels != initialPixels) {
+			DrawCutsceneBackground();
+			if (DiabloUiSurface() == PalSurface)
+				BltFast(nullptr, nullptr);
+			RenderPresent();
+		}
+	}
+	FreeCutsceneBackground();
+
+	// The loading thread sets `logical_palette`, so we make sure to use
+	// our own palette for the fade-in.
+	PaletteFadeIn(8, ProgressEventHandlerState.palette);
+}
+
+void CheckShouldSkipRendering()
+{
+	if (!ProgressEventHandlerState.skipRendering) return;
+	const bool shouldSkip = ProgressEventHandlerState.loadStartedAt + *GetOptions().Gameplay.skipLoadingScreenThresholdMs > SDL_GetTicks();

```

Comment:
Does this really make sense as an option it seems more of a baked in platform value

Reply:
Note that this PR does not change the `CheckShouldSkipRendering` at all -- it simply moves it around.